File: src/ast/CastExpression.java
package ast;

import java.io.StringWriter;

public class CastExpression extends BaseNode {
    private String _kind;
    private BaseNode _to;
    private BaseNode _from;

    public CastExpression(String kind, BaseNode to, BaseNode from) {
        super(NodeType.CastExpression);
        _kind = kind;
        _to = to;
        _from = from;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        writer.write(_kind);
        writer.write("<");
        _to.PrintLeft(writer);
        writer.write(">(");
        _from.PrintLeft(writer);
        writer.write(")");
    }
}

File: src/de/fearlesstobi/demangler/ast/CastExpression.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class CastExpression(private val kind: String, private val to: BaseNode, private val from: BaseNode) : BaseNode(NodeType.CastExpression) {
    public override fun printLeft(writer: StringWriter) {
        writer.write(kind)
        writer.write("<")
        to.printLeft(writer)
        writer.write(">(")
        from.printLeft(writer)
        writer.write(")")
    }

}